src/darwin/Framework/CHIP/CHIPDeviceController+XPC.h

```diff
@@ -35,6 +35,28 @@ NS_ASSUME_NONNULL_BEGIN
 + (CHIPDeviceController *)sharedControllerWithId:(id<NSCopying> _Nullable)controllerId
                                  xpcConnectBlock:(NSXPCConnection * (^)(void) )connectBlock;
 
+/**
+ * Returns a serialized values object to send over XPC for read, write and command interactions
+ */
++ (NSArray<NSDictionary<NSString *, id> *> * _Nullable)serializeXPCResponseValues:
+    (NSArray<NSDictionary<NSString *, id> *> * _Nullable)values;
+
+/**
+ * Returns a deserialized values object from a values object received from XPC for read, write and command interactions
+ */
++ (NSArray<NSDictionary<NSString *, id> *> * _Nullable)deserializeXPCResponseValues:
+    (NSArray<NSDictionary<NSString *, id> *> * _Nullable)values;
+
+/**
+ * Returns a serialized value object to send over XPC as report
+ */
++ (NSDictionary<NSString *, id> * _Nullable)serializeXPCReportValue:(NSDictionary<NSString *, id> * _Nullable)value;
```

Comment:
Why "ResponseValues" but "ReportValue"?

Reply:
I could obsolete ReportValue per consolidating report with CHIPDeviceResponseHandler. The original reason they were separate was that report handler receives a single value while response for read, write, invoke command potentially receive multiple values. Such separation probably doesn't provide much benefit to app writers.

Reply:
I'll also rename the serialize / deserialize to encode / decode.